jangjuseong/app.js

```diff
@@ -50,12 +50,147 @@ app.post('/users/signup', async (req, res) => {
         ?, 
         ?, 
         ?
-        );
+      );
     `,
     [name, email, password, profileImage]
   );
 
-  res.status(201).json({ message: 'userCreated' });
+  return res.status(201).json({ message: 'userCreated' });
+});
+
+app.post('/posts', async (req, res) => {
+  const { title, content, postImgUrl, userId } = req.body;
+
+  await mysqlDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      post_image_url,
+      user_id
+      )
+      VALUES (
+        ?,
+        ?,
+        ?,
+        ?
+      );
+        `,
+    [title, content, postImgUrl, userId]
+  );
+
+  return res.status(201).json({ message: 'postCreated' });
+});
+
+app.get('/posts', async (req, res) => {
+  const posts = await mysqlDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts p
+    INNER JOIN users u ON u.id = p.user_id;
+    `
+  );
+
+  return res.status(200).json({ posts });
+});
+
+app.get('/posts/:userId', async (req, res) => {
+  const { userId } = req.params;
+
+  const [result] = await mysqlDataSource.query(
+    `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      pi.post_informations AS postings
+    FROM users u
+    INNER JOIN (
+      SELECT
+        user_id,
+        JSON_ARRAYAGG(
+          JSON_OBJECT (
+            "postingId", id,
+            "postingImageUrl", post_image_url,
+            "postingContent", content
+          )
+        ) AS post_informations
+      FROM posts
+      GROUP BY user_id
+    ) pi
+      ON pi.user_id = u.id
+    WHERE u.id = ?;
+    `,
+    [userId]
+  );
+
+  return res.status(200).json({ data: result });
+});
+
+app.patch('/posts/:postId/:userId', async (req, res) => {
+  const { userId, postId } = req.params;
+  const { content } = req.body;
+
+  await mysqlDataSource.query(
+    `UPDATE posts
+    SET
+      content = ?
+    WHERE 
+      user_id = ? AND id = ?
+    `,
+    [content, userId, postId]
+  );
+
+  const [result] = await mysqlDataSource.query(
+    `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts p
+    INNER JOIN users u ON u.id = p.user_id
+    WHERE u.id = ? AND p.id = ?
+    `,
+    [userId, postId]
+  );
+
+  return res.status(201).json({ data: result });
+});
+
+app.delete('/posts/:postId', async (req, res) => {
+  const { postId } = req.params;
+
+  await mysqlDataSource.query(
+    `DELETE
+    FROM posts
+    WHERE posts.id = ${postId}
```

Comment:
`?` 매핑을 통해 데이터를 삭제할 수 있도록 수정해주세요!

```suggestion
    WHERE posts.id = ?
```